graspy/inference/latent_distribution_test.py

```diff
@@ -141,27 +180,40 @@ def fit(self, A1, A2):
 
         Returns
         -------
-        p_ : float
+        p_value : float
             The p value corresponding to the specified hypothesis test
         """
         A1 = import_graph(A1)
         A2 = import_graph(A2)
-        # if not is_symmetric(A1) or not is_symmetric(A2):
-        #     raise NotImplementedError()  # TODO asymmetric case
-        if self.n_components is None:
-            # get the last elbow from ZG for each and take the maximum
-            num_dims1 = select_dimension(A1)[0][-1]
-            num_dims2 = select_dimension(A2)[0][-1]
-            self.n_components = max(num_dims1, num_dims2)
 
         X1_hat, X2_hat = self._embed(A1, A2)
-        X1_hat, X2_hat = self._median_heuristic(X1_hat, X2_hat)
-        U = self._statistic(X1_hat, X2_hat)
-        null_distribution = self._bootstrap(X1_hat, X2_hat, self.n_bootstraps)
-        self.null_distribution_ = null_distribution
-        self.sample_T_statistic_ = U
-        p_value = (len(null_distribution[null_distribution >= U])) / self.n_bootstraps
-        if p_value == 0:
-            p_value = 1 / self.n_bootstraps
-        self.p_ = p_value
-        return self.p_
+        X1_hat, X2_hat = _median_sign_flips(X1_hat, X2_hat)
+
+        data = self.test.test(
+            X1_hat, X2_hat, reps=self.n_bootstraps, workers=self.workers, auto=False
```

Comment:
@bdpedigo I made the auto=False yesterday (forces the tests to be permutation, as opposed to using Chi-squared approximations, when possible), but I wanted to run that by you. It seems to me that it is better to use maybe computationally expensive permutation test that we know the behavior of, rather than using approximations that were never proven for graphs, and might not even be provable. this is not relevant for some tests like mgc, which is always permutation based, but is relevant for, say, dCorr. we could also just make this an argument. in any case - your call.

Reply:
let's just get this merged as is - on one hand I think the chi-squared is supposed to be conservative if anything after a relatively small number of samples. On the other hand, yes we have not tested, and we've been wrong about this stuff before... so I'm okay with having it not be an option for now but probably worth (someone) looking into

Reply:
agreed